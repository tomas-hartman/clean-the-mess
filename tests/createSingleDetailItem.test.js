const { createSingleDetailItem } = require("../src/popup/popup");

it('should render correct detail element for latest', () => {
    const data = {
        date: "18. 5. 2020 21:45:35",
        title: "Youtube",
        id: 65,
        url: "https://www.youtube.com/"
    }
    const props = {
        data,
        itemId: 1,
        type: "latest"
    }
    const output = createSingleDetailItem(props);

    expect(output).toMatchSnapshot();
});

it('should render correct detail element for details', () => {
    const data = {
        "id": 23,
        "index": 2,
        "windowId": 1,
        "highlighted": false,
        "active": false,
        "attention": false,
        "pinned": false,
        "status": "complete",
        "hidden": false,
        "discarded": false,
        "incognito": false,
        "width": 862,
        "height": 857,
        "lastAccessed": 1589732493951,
        "audible": false,
        "mutedInfo": {
          "muted": false
        },
        "isInReaderMode": false,
        "sharingState": {
          "camera": false,
          "microphone": false
        },
        "successorTabId": -1,
        "url": "https://www.respekt.cz/tydenik/2009/19/vezeni-gaza",
        "title": "Vězení Gaza • RESPEKT",
        "favIconUrl": "data:image/x-icon;base64"
      }

    const props = {
        data,
        itemId: 1,
        type: "details"
    }
    const output = createSingleDetailItem(props);

    expect(output).toMatchSnapshot();
});