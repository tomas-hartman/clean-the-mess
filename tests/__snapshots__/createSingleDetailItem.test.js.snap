// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct detail element for details 1`] = `
<DocumentFragment>
  
        
  <li
    class="detail"
    data-tab-id="23"
    id="item-1"
  >
    
            
    <div
      class="item-container detail"
    >
      
                
      <div
        class="item-text-container"
      >
        
                    
        <div
          class="title detail"
          title="Vězení Gaza • RESPEKT"
        >
          Vězení Gaza • RESPEKT
        </div>
        
                    
        <div
          class="url detail "
          title="https://www.respekt.cz/tydenik/2009/19/vezeni-gaza"
        >
          https://www.respekt.cz/tydenik/2009/19/vezeni-gaza
        </div>
        
                    
        <div
          class="last-displayed detail hidden"
          title="undefined"
        >
          undefined
        </div>
        
                
      </div>
      
                
      <div
        class="item-buttons-container"
      >
        
                    
        <div
          class="bookmark bookmark-close detail hidden"
          title="Bookmark and close tab"
        />
        
                    
        <div
          class="remove detail hidden"
          title="Close tab"
        />
        
                    
        <div
          class="get-in"
        />
        
                
      </div>
      
            
    </div>
    
        
  </li>
  
        
</DocumentFragment>
`;

exports[`should render correct detail element for latest 1`] = `
<DocumentFragment>
  
        
  <li
    class="detail"
    data-tab-id="65"
    id="item-1"
  >
    
            
    <div
      class="item-container detail"
    >
      
                
      <div
        class="item-text-container"
      >
        
                    
        <div
          class="title detail"
          title="Youtube"
        >
          Youtube
        </div>
        
                    
        <div
          class="url detail hidden"
          title="https://www.youtube.com/"
        >
          https://www.youtube.com/
        </div>
        
                    
        <div
          class="last-displayed detail "
          title="18. 5. 2020 21:45:35"
        >
          18. 5. 2020 21:45:35
        </div>
        
                
      </div>
      
                
      <div
        class="item-buttons-container"
      >
        
                    
        <div
          class="bookmark bookmark-close detail hidden"
          title="Bookmark and close tab"
        />
        
                    
        <div
          class="remove detail hidden"
          title="Close tab"
        />
        
                    
        <div
          class="get-in"
        />
        
                
      </div>
      
            
    </div>
    
        
  </li>
  
        
</DocumentFragment>
`;
